templates/essentials.yaml

```diff
@@ -466,6 +492,16 @@ Outputs:
     Value: !Ref AWSKmsInfraKeyAlias
     Export:
       Name: !Sub '${AWS::Region}-${AWS::StackName}-InfraKeyAlias'
+  ManagedInstanceProfile:
+    Description: Profile to allow instances to tag its root volume and allow SSM actions
+    Value: !Ref ManagedInstanceProfile
+    Export:
+      Name: !Sub '${AWS::Region}-${AWS::StackName}-ManagedInstanceProfile'
+  ManagedInstanceRole:
+    Description: Role to allow instances to tag its root volume
+    Value: !Ref ManagedInstanceRole
+    Export:
+      Name: !Sub '${AWS::Region}-${AWS::StackName}-ManagedInstanceRole'
```

Comment:
These are outputs of the logical names of the resource however ARNs typically useful than names.  I would suggest either replacing these outputs with the ARN or adding outputs for ARNs.  for example..

```
  ManagedInstanceRoleArn:
    Description: Role ARN to allow instances to tag its root volume
    Value: !GetAtt ManagedInstanceRole.Arn
    Export:
      Name: !Sub '${AWS::Region}-${AWS::StackName}-ManagedInstanceRoleArn'
```

Reply:
OK, I've add ARN strings as exports for some resources. I wonder if we would hit a limit doubling up on exports.